include/options/option/abstract-option.php

```diff
@@ -0,0 +1,316 @@
+		foreach ( $this->errors->get_error_codes() as $code ) {
+			$data = $this->errors->get_error_data( $code );
+
+			if ( empty( $data ) || 'error' === $data ) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Validates option's value, can be overridden for specific cases not handled by `rest_validate_value_from_schema`.
+	 * If the validation fails, the value must be rejected.
+	 *
+	 * @since 3.7
+	 *
+	 * @param mixed   $value   Value to validate.
+	 * @param Options $options All options.
+	 * @return bool True on success, false otherwise.
+	 */
+	protected function validate( $value, Options $options ): bool { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$is_valid = rest_validate_value_from_schema( $value, $this->get_schema(), $this->key() );
+
+		if ( is_wp_error( $is_valid ) ) {
+			// Invalid: blocking error.
+			$this->errors->merge_from( $this->make_error_unique( $is_valid ) );
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Sanitizes option's value, can be overridden for specific cases not handled by `rest_sanitize_value_from_schema()`.
+	 * Can populate the `$errors` property with blocking and non-blocking errors: in case of non-blocking errors,
+	 * the value is sanitized and can be stored.
+	 *
+	 * @since 3.7
+	 *
+	 * @param mixed   $value   Value to sanitize.
+	 * @param Options $options All options.
+	 * @return mixed The sanitized value. The previous value in case of blocking error.
+	 */
+	protected function sanitize( $value, Options $options ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$value = rest_sanitize_value_from_schema( $value, $this->get_schema(), $this->key() );
+
+		if ( is_wp_error( $value ) ) {
+			// Blocking error.
+			$this->errors->merge_from( $this->make_error_unique( $value ) );
+			return $this->value;
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Creates JSON schema of the option.
+	 *
+	 * @since 3.7
+	 *
+	 * @return array The schema.
+	 */
+	abstract protected function create_schema(): array;
+
+	/**
+	 * Returns a base for a JSON schema of the option.
+	 *
+	 * @since 3.7
+	 *
+	 * @param array $schema A list of data to add to the schema. At least the key `type` must be added.
+	 * @return array The schema.
+	 *
+	 * @phpstan-param array{type: SchemaType}&array<non-falsy-string, mixed> $schema
+	 * @phpstan-return Schema
+	 */
+	protected function build_schema( array $schema ): array {
+		return array_merge(
+			array(
+				'$schema'     => 'http://json-schema.org/draft-04/schema#',
+				'title'       => $this->key(),
+				'description' => $this->description,
+				'context'     => array( 'edit' ),
+			),
+			$schema
+		);
+	}
+
+	/**
+	 * Changes error codes so they are unique to the option.
+	 * Copied from `WP_Error::copy_errors()`.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_Error $errors An error object.
+	 * @return WP_Error
+	 */
+	protected function make_error_unique( WP_Error $errors ): WP_Error {
```

Comment:
So error codes returned by `rest_validate_value_from_schema()` and `rest_sanitize_value_from_schema()` are unique to each option.

Reply:
Removed in [8ce014a](https://github.com/polylang/polylang/pull/1451/commits/8ce014a391a03a41839f1f8079f834580245fed8): with the new way of handling errors, we can't differentiate errors coming from `rest_validate_value_from_schema()` and `rest_sanitize_value_from_schema()` (from the outside).
Note: it could still be used inside `validate()` and `sanitize()`.
Note 2: even with the same error code, messages will simply stack.